tests/actions/reposync_test.py

```diff
@@ -0,0 +1,251 @@
+        input_mirror_type,
+        input_mirror,
+        input_arch,
+        input_rpm_list,
+        expected_exception,
+        api,
+        repo,
+        reposync
+    ):
+        # Arrange
+        test_repo = repo
+        test_repo.breed = enums.RepoBreeds.APT
+        test_repo.arch = input_arch
+        test_repo.apt_components = "main"
+        test_repo.apt_dists = "stable"
+        test_repo.mirror = input_mirror
+        test_repo.mirror_type = input_mirror_type
+        test_repo.rpm_list = input_rpm_list
+        test_repo.yumopts = "--exclude=.* --include=dpkg.* --no-check-gpg --rsync-extra=none"
+        test_settings = api.settings()
+        repo_path = os.path.join(test_settings.webdir, "repo_mirror", test_repo.name)
+
+        # Act & Assert
+        with expected_exception:
+            reposync.run(test_repo.name)
+            result = os.path.exists(repo_path)
+            for rpm in ["dpkg"]:
+                assert glob.glob(os.path.join(repo_path, "**", "dpkg") + "*", recursive=True) != []
+            assert result
+            reposync.run(test_repo.name)
+
+    @pytest.mark.usefixtures("remove_repo")
+    @pytest.mark.parametrize(
+        "input_mirror_type,input_mirror,expected_exception",
+        [
+            (
+                enums.MirrorType.BASEURL,
+                "http://download.fedoraproject.org/pub/fedora/linux/releases/35/Everything/x86_64/os/Packages/2",
+                does_not_raise()
+            ),
+            (
+                enums.MirrorType.MIRRORLIST,
+                "http://download.fedoraproject.org/pub/fedora/linux/releases/35/Everything/x86_64/os/Packages/2",
+                pytest.raises(cexceptions.CX)
+            ),
+            (
+                enums.MirrorType.METALINK,
+                "http://download.fedoraproject.org/pub/fedora/linux/releases/35/Everything/x86_64/os/Packages/2",
+                pytest.raises(cexceptions.CX)
+            ),
+            (
+                enums.MirrorType.BASEURL,
+                "http://www.example.com/path/to/some/repo",
+                pytest.raises(cexceptions.CX)
+            ),
+        ],
+    )
+    def test_reposync_wget(
+        self,
+        input_mirror_type,
+        input_mirror,
+        expected_exception,
+        api,
+        repo,
+        reposync
+    ):
+        # Arrange
+        test_repo = repo
+        test_repo.breed = enums.RepoBreeds.WGET
+        test_repo.mirror = input_mirror
+        test_repo.mirror_type = input_mirror_type
+        test_settings = api.settings()
+        repo_path = os.path.join(test_settings.webdir, "repo_mirror", test_repo.name)
+
+        # Act & Assert
+        with expected_exception:
+            reposync.run(test_repo.name)
+            result = os.path.exists(repo_path)
+            for rpm in ["rpm"]:
+                assert glob.glob(os.path.join(repo_path, "**", "2") + "*", recursive=True) != []
+            assert result
+            reposync.run(test_repo.name)
```

Comment:
Same question as above. Second `reposync.run()`?

Reply:
@SchoolGuy There is no need for this code here. I will remove it.